tests/scattering1d/test_timefrequency_scattering.py

```diff
@@ -59,6 +59,63 @@ def test_Q():
     assert "Q_fr must be an integer or 1-tuple" in ve.value.args[0]
 
 
+def test_check_runtime_args():
+    kwargs = dict(J=10, J_fr=3, shape=4096, Q=8)
+    S = TimeFrequencyScatteringBase(**kwargs)
+    S.build()
+    assert S._check_runtime_args() is None
+
+    with pytest.raises(ValueError) as ve:
+        S = TimeFrequencyScatteringBase(out_type='doesnotexist', **kwargs)
+        S.build()
+        S._check_runtime_args()
+    assert "out_type must be one of" in ve.value.args[0]
+
+    with pytest.raises(ValueError) as ve:
+        S = TimeFrequencyScatteringBase(T=0, **kwargs)
+        S.build()
+        S._check_runtime_args()
+    assert "Cannot convert" in ve.value.args[0]
+
+    with pytest.raises(ValueError) as ve:
+        S = TimeFrequencyScatteringBase(oversampling=-1, **kwargs)
+        S.build()
+        S._check_runtime_args()
+    assert "nonnegative" in ve.value.args[0]
+
+    with pytest.raises(ValueError) as ve:
+        S = TimeFrequencyScatteringBase(oversampling=0.5, **kwargs)
+        S.build()
+        S._check_runtime_args()
+    assert "integer" in ve.value.args[0]
+
+    with pytest.raises(ValueError) as ve:
+        S = TimeFrequencyScatteringBase(
+            F=0, out_type='array', format='joint', **kwargs)
+        S.build()
+        S._check_runtime_args()
+    assert "Cannot convert" in ve.value.args[0]
+
+    with pytest.raises(ValueError) as ve:
+        S = TimeFrequencyScatteringBase(oversampling_fr=-1, **kwargs)
+        S.build()
+        S._check_runtime_args()
+    assert "nonnegative" in ve.value.args[0]
+
+    with pytest.raises(ValueError) as ve:
+        S = TimeFrequencyScatteringBase(oversampling_fr=0.5, **kwargs)
+        S.build()
+        S._check_runtime_args()
+    assert "integer" in ve.value.args[0]
```

Comment:
line 80-90 is the same as line 99-109.

Reply:
not exactly. Lines 80-90 test `oversampling`, lines 99-109 test `oversampling_fr`

Reply:
you are right, my bad!